tf_agents/agents/dqn/examples/v1/train_eval_gym.py

```diff
@@ -253,6 +255,9 @@ def train_eval(
 
         global_step_val = global_step_call()
 
+	if global_step_val % epsilon_decay_period == 0 and tf_agent.epsilon_greedy > min_epsilon:
+	  tf_agent.epsilon_greedy *= epsilon_decay
```

Comment:
You cannot do this in TF v1, you need to use a tf.Variable to represent the epsilon_greedy and updated using the global_step.

Reply:
Do you want me to add the following:
Declare epsilon_greedy as tf.Variable
`epsilon_greedy=tf.Variable(1.0, dtype=tf.float64)`

Epsilon decay tf function
```
@tf.function
def decay_epsilon(global_step_val, epsilon_greedy=tf_agent.epsilon_greedy, epsilon_decay_period=epsilon_decay_period, min_epsilon=min_epsilon):
    if global_step_val % epsilon_decay_period == 0 and epsilon_greedy > min_epsilon:
        epsilon_greedy *= epsilon_decay
    return epsilon_greedy
```
Update epsilon
`tf_agent.epsilon_greedy = decay_epsilon(global_step_val)`